analyses/org.eclipse.tracecompass.incubator.executioncomparision.core/src/org/eclipse/tracecompass/incubator/internal/executioncomparison/core/DifferentialSeqCallGraphAnalysis.java

```diff
@@ -0,0 +1,409 @@
+        }
+        List<ICallStackElement> leafGroups = new ArrayList<>();
+        group.getChildrenElements().forEach(g -> leafGroups.addAll(getLeafElement(g)));
+        return leafGroups;
+    }
+
+    private static void recurseAddElementData(ICallStackElement element, String group, CallGraph callGraph, WeightedTreeSet<ICallStackSymbol, Object> newTreeSet) {
+        // Add the current level of trees to the new tree set
+        for (AggregatedCallSite tree : callGraph.getCallingContextTree(element)) {
+            newTreeSet.addWeightedTree(group, tree.copyOf());
+        }
+
+        // Recursively add the next level of elements
+        ICallStackElement treeEl = element;
+        Collection<ITree> children = treeEl.getChildren();
+        for (ITree child : children) {
+            if (child instanceof ICallStackElement) {
+                recurseAddElementData((ICallStackElement) child, group, callGraph, newTreeSet);
+            }
+        }
+    }
+
+    @Override
+    protected boolean executeAnalysis(IProgressMonitor monitor) throws TmfAnalysisException {
+        return true;
+    }
+
+    @Override
+    public boolean waitForCompletion() {
+        return true;
+
+    }
+
+    @Override
+    protected void canceling() {
+        // Does nothing
+    }
+
+    /**
+     * Update the selection range
+     *
+     * @param signal
+     *            TmfComparisonFilteringUpdatedSignal raised when filtering
+     *            parameter are changed
+     *
+     */
+    @TmfSignalHandler
+    public void selectionRangeUpdated(TmfComparisonFilteringUpdatedSignal signal) {
+        // tuning start and end times
+        fStartA = TmfTimestamp.BIG_BANG.equals(signal.getBeginTimeA()) ? fStartA : signal.getBeginTimeA();
+        fEndA = TmfTimestamp.BIG_CRUNCH.equals(signal.getEndTimeA()) ? fEndA : signal.getEndTimeA();
+        fStartB = TmfTimestamp.BIG_BANG.equals(signal.getBeginTimeB()) ? fStartB : signal.getBeginTimeB();
+        fEndB = TmfTimestamp.BIG_CRUNCH.equals(signal.getEndTimeB()) ? fEndB : signal.getEndTimeB();
+        // tuning fStatistic
+        String statistic = signal.getStatistic();
+        fStatistic = (statistic == null) ? fStatistic : statistic;
+        // tuning fTraceList
+        List<String> traceListA = signal.getTraceListA();
+        if (traceListA != null) {
+            synchronizedListAdd(traceListA, fTraceListA);
+
+        }
+        List<String> traceListB = signal.getTraceListB();
+        if (traceListB != null) {
+            synchronizedListAdd(traceListB, fTraceListB);
+
+        }
+        if (!fTraceCallGraphRegistry.isEmpty()) {
+            try (ScopeLog sl = new ScopeLog(LOGGER, Level.FINE, "MakeDiffCallGraph")) { //$NON-NLS-1$
+                synchronized (this) {
+                    if (fDiffJob != null) {
+                        fDiffJob.cancel();
+                    }
+                    fDiffJob = new Job("Make differential Callgraph") { //$NON-NLS-1$
```

Comment:
This job should be part of the executeAnalysis mechanism and other inherited methods from TmfAbstractAnalysisModule.